src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -117,6 +117,13 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
                     {
                         req.Headers.Append(header.Key, string.Join(" ", header.Value));
                     }
+                    else if (string.Equals(header.Key, HeaderNames.ContentLength, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (long.TryParse(header.Value.First(), out var contentLength))
+                        {
+                            req.ContentLength = contentLength;
+                        }
+                    }
```

Comment:
This shouldn't have affected the behavior, the Asp.Net ContentLength property is already read from the header. You should get the same test results if you remove this section. Does the test fail before adding this and pass after?

The expected workaround was reading System.Net.Http's ContentLength property because it has the side-effect of generating the header. https://github.com/dotnet/aspnetcore/issues/18463#issuecomment-613804982

Reply:
> Does the test fail before adding this and pass after? 

Yes, but I changed the code as you said. Reading the property would set the header. Now if there's no body on the request the `ContentLength` is Zero. Is this the expected behavior?